### PR TITLE
Split signers to prepush and non prepush

### PR DESCRIPTION
**Pull request by midnightercz**
Cosign signer cannot work without image being available in destination registry. Therefore signer wrappers now have attribute pre_push which indicates if signer should run before container push stage and which after

### CODE REVIEW

**Review by querti — CHANGES_REQUESTED** [review]
What about index images? Does signing work if they are not pushed?

File: pubtools/_quay/iib_operations.py [hunk 1]
```diff
@@ -169,8 +170,18 @@ def _sign_index_image(
     current_signatures: list[tuple[str, str, str]] = [
         (e.reference, e.digest, e.signing_key) for e in to_sign_entries
     ]
+    if pre_push:
```
Comment (querti) [inline pubtools/_quay/iib_operations.py:173]: Should this be here?

**Review by querti — CHANGES_REQUESTED** [review]

File: pubtools/_quay/iib_operations.py (hunk 1 above)
Comment (midnightercz) [reply pubtools/_quay/iib_operations.py:173]: you mean if it these methods shouldn't be available on the global level?

File: pubtools/_quay/iib_operations.py [hunk 2]
```diff
@@ -150,6 +150,7 @@ def _sign_index_image(
     signing_keys: list[str],
     task_id: str,
     target_settings: dict[str, Any],
+    pre_push: bool = False,
```
Comment (emilyzheng) [inline pubtools/_quay/iib_operations.py:153]: Add to docstring?

File: pubtools/_quay/iib_operations.py (hunk 1 above)
Comment (querti) [reply pubtools/_quay/iib_operations.py:173]: I'm trying to understand this code. My best guess: if pre_push is true, only use prepush signers, if it's false only  use pust push signers. If that's so, why not just:
SIGNER_BY_LABEL[signer["label"]].pre_push == pre_push

Comment (midnightercz) [reply pubtools/_quay/iib_operations.py:173]: yeah, it can be simplified like that

**Review by querti — APPROVED** [review]

**Review by emilyzheng — APPROVED** [review]

**Pull request merged**